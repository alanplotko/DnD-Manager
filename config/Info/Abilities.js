export default [
  'strength', 'dexterity', 'constitution', 'intelligence', 'wisdom', 'charisma',
];
