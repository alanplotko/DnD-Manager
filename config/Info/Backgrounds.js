export default [
  {
    key: 'acolyte',
    name: 'Acolyte',
    description: 'You have spent your life in the service of a temple to a specific god or pantheon of gods. You act as an intermediary between the realm of the holy and the mortal world, performing sacred rites and offering sacrifices in order to conduct worshipers into the presence of the divine.',
    starting: {
      decisions: [
        [
          {
            name: 'Prayer book',
            description: null,
            quantity: 1,
            unit: null,
          },
          {
            name: 'Prayer wheel',
            description: null,
            quantity: 1,
            unit: null,
          },
        ],
      ],
      equipment: [
        {
          name: 'Holy symbol',
          description: 'A gift to you when you entered the priesthood.',
          quantity: 1,
          unit: null,
        },
        {
          name: 'Stick of incense',
          description: null,
          quantity: 5,
          unit: { singular: 'stick', plural: 'sticks' },
        },
        {
          name: 'Vestments',
          description: null,
          quantity: null,
          unit: null,
        },
        {
          name: 'Common clothes',
          description: null,
          quantity: 1,
          unit: { singular: 'set', plural: 'sets' },
        },
      ],
      money: {
        gp: 15,
      },
    },
    additionalLanguages: 2,
    proficiencies: {
      skills: ['insight', 'religion'],
      skillKeys: ['insight', 'religion'],
      tools: [],
    },
  },
];
