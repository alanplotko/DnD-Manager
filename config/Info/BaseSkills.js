export default {
  acrobatics: {
    skillLabel: 'Acrobatics',
    modifier: 0,
    ability: 'dexterity',
    proficient: false,
  },
  animalHandling: {
    skillLabel: 'Animal Handling',
    modifier: 0,
    ability: 'wisdom',
    proficient: false,
  },
  arcana: {
    skillLabel: 'Arcana',
    modifier: 0,
    ability: 'intelligence',
    proficient: false,
  },
  athletics: {
    skillLabel: 'Athletics',
    modifier: 0,
    ability: 'strength',
    proficient: false,
  },
  deception: {
    skillLabel: 'Deception',
    modifier: 0,
    ability: 'charisma',
    proficient: false,
  },
  history: {
    skillLabel: 'History',
    modifier: 0,
    ability: 'intelligence',
    proficient: false,
  },
  insight: {
    skillLabel: 'Insight',
    modifier: 0,
    ability: 'wisdom',
    proficient: false,
  },
  intimidation: {
    skillLabel: 'Intimidation',
    modifier: 0,
    ability: 'charisma',
    proficient: false,
  },
  investigation: {
    skillLabel: 'Investigation',
    modifier: 0,
    ability: 'intelligence',
    proficient: false,
  },
  medicine: {
    skillLabel: 'Medicine',
    modifier: 0,
    ability: 'wisdom',
    proficient: false,
  },
  nature: {
    skillLabel: 'Nature',
    modifier: 0,
    ability: 'intelligence',
    proficient: false,
  },
  perception: {
    skillLabel: 'Perception',
    modifier: 0,
    ability: 'wisdom',
    proficient: false,
  },
  performance: {
    skillLabel: 'Performance',
    modifier: 0,
    ability: 'charisma',
    proficient: false,
  },
  persuasion: {
    skillLabel: 'Persuasion',
    modifier: 0,
    ability: 'charisma',
    proficient: false,
  },
  religion: {
    skillLabel: 'Religion',
    modifier: 0,
    ability: 'intelligence',
    proficient: false,
  },
  sleightOfHand: {
    skillLabel: 'Sleight of Hand',
    modifier: 0,
    ability: 'dexterity',
    proficient: false,
  },
  stealth: {
    skillLabel: 'Stealth',
    modifier: 0,
    ability: 'dexterity',
    proficient: false,
  },
  survival: {
    skillLabel: 'Survival',
    modifier: 0,
    ability: 'wisdom',
    proficient: false,
  },
};
