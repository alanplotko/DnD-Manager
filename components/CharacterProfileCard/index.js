import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';
import { Card, CardItem, Text, Body, Left, Right } from 'native-base';
import { COLOR, Icon, IconToggle, ListItem, ThemeContext } from 'react-native-material-ui';
import { StyleSheet, Image, View } from 'react-native';
import * as Progress from 'react-native-progress';
import { EXPERIENCE, IMAGES } from 'FifthEditionManager/config/Info';
import { getCharacterDisplayName } from 'FifthEditionManager/util';

const calculateLevelProgress = (level, experience) => (
  level < 20 ? experience / EXPERIENCE[level] : 1
);
const formatNumber = num => (
  num > 999 ? `${(num / 1000).toFixed(1)}k` : num
);
const navItem = {
  container: {
    margin: 0,
    height: 50,
  },
  leftElementContainer: {
    marginRight: -10,
  },
};
const navHeader = Object.assign({}, navItem, {
  container: {
    margin: 0,
    borderBottomWidth: 1,
    borderBottomColor: COLOR.grey300,
    height: 55,
  },
  contentViewContainer: {
    backgroundColor: COLOR.grey50,
  },
});
const navText = {
  fontFamily: 'Roboto',
  fontSize: 14,
  color: COLOR.black,
};

const CharacterProfileCard = (props) => {
  const modalContent = (
    <ThemeContext.Consumer>
      {theme => (
        <View style={{ margin: 0 }}>
          <ListItem
            leftElement={
              <Image
                source={IMAGES.RACE[props.character.race.lookupKey]}
                style={{ height: 36, width: 36 }}
                resizeMode="contain"
              />
            }
            centerElement={
              <Text style={[styles.characterName, { color: theme.palette.textColor }]}>
                {getCharacterDisplayName(props.character)}
              </Text>
            }
            style={navHeader}
          />
          <ListItem
            leftElement={<Icon name="open-in-new" color={theme.palette.iconColor} />}
            centerElement={
              <Text style={navText}>View Character</Text>
            }
            onPress={() => props.viewHandler(props.character.key)}
            style={navItem}
          />
          <ListItem
            leftElement={<Icon name="mode-edit" color={theme.palette.iconColor} />}
            centerElement={
              <Text style={navText}>Edit Character</Text>
            }
            onPress={() => props.editHandler(props.character.key)}
            style={navItem}
          />
          <ListItem
            leftElement={<Icon name="delete" color={theme.palette.iconColor} />}
            centerElement={
              <Text style={navText}>Delete Character</Text>
            }
            onPress={() => props.deleteHandler(props.character)}
            style={navItem}
          />
        </View>
      )}
    </ThemeContext.Consumer>
  );

  return (
    <ThemeContext.Consumer>
      {theme => (
        <Card style={{ marginLeft: 10, marginRight: 10, marginTop: 10 }}>
          <CardItem cardBody>
            <Image
              source={IMAGES.RACE[props.character.race.lookupKey]}
              style={{ height: 100, width: null, flex: 1 }}
            />
            <View style={{ position: 'absolute', top: 0, right: 0 }}>
              <IconToggle
                name="more-vert"
                color={COLOR.white}
                onPress={() => props.modalHandler(modalContent)}
              />
            </View>
            <View style={{ position: 'absolute', bottom: 0, right: 0 }}>
              <Text
                style={[
                  styles.lastUpdatedBanner,
                  { backgroundColor: theme.palette.fadedBackgroundColor },
                ]}
              >
                Last updated {moment(props.character.meta.lastUpdated).fromNow()}
              </Text>
            </View>
          </CardItem>
          <CardItem>
            <Left style={{ flex: 2 }}>
              <Image
                source={IMAGES.BASE_CLASS.ICON[props.character.baseClass.lookupKey]}
                style={{ width: 48, height: 64 }}
                resizeMode="contain"
              />
              <Body>
                <Text
                  style={[
                    styles.heading,
                    { color: theme.palette.textColor },
                  ]}
                  numberOfLines={1}
                >
                  {getCharacterDisplayName(props.character)}&nbsp;
                </Text>
                <Text style={[styles.subheading, { color: theme.palette.noteColor }]}>
                  {props.character.race.name}&nbsp;
                  {props.character.baseClass.name}
                </Text>
                <Text style={[styles.subheading, { color: theme.palette.noteColor }]}>
                  {props.character.background.name}
                </Text>
              </Body>
            </Left>
            <Right style={{ flex: 1 }}>
              <View
                style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}
              >
                <Text style={[styles.levelText, { color: theme.palette.noteColor }]}>
                  Level {props.character.profile.level}
                </Text>
                <Progress.Bar
                  progress={calculateLevelProgress(
                    props.character.profile.level,
                    props.character.profile.experience,
                  )}
                  animated={false}
                  color={theme.palette.primaryColor}
                  borderColor={theme.palette.primaryColor}
                  width={100}
                  style={{ marginTop: 3, marginBottom: 3 }}
                />
                {
                  props.character.profile.level < 20 &&
                  <Text style={styles.experienceText}>
                    {formatNumber(props.character.profile.experience)} /&nbsp;
                    {formatNumber(EXPERIENCE[props.character.profile.level])}
                  </Text>
                }
                {
                  props.character.profile.level === 20 &&
                  <Text style={styles.experienceText}>
                    {formatNumber(props.character.profile.experience)}
                  </Text>
                }
              </View>
            </Right>
          </CardItem>
        </Card>
      )}
    </ThemeContext.Consumer>
  );
};

const styles = StyleSheet.create({
  heading: {
    fontFamily: 'RobotoLight',
    fontSize: 18,
    color: COLOR.black,
    width: 175,
  },
  subheading: {
    fontFamily: 'Roboto',
    fontSize: 14,
    color: COLOR.grey500,
  },
  characterName: {
    fontFamily: 'Roboto',
    fontSize: 14,
    color: COLOR.black,
  },
  levelText: {
    fontFamily: 'RobotoLight',
    fontSize: 16,
    color: COLOR.grey500,
  },
  experienceText: {
    fontFamily: 'RobotoLight',
    fontSize: 14,
    color: COLOR.grey500,
  },
  lastUpdatedBanner: {
    paddingLeft: 5,
    paddingRight: 5,
    paddingTop: 3,
    paddingBottom: 3,
    color: COLOR.white,
    fontFamily: 'Roboto',
    fontStyle: 'italic',
    fontSize: 12,
    borderTopLeftRadius: 5,
  },
});

CharacterProfileCard.propTypes = {
  character: PropTypes.object.isRequired,
  modalHandler: PropTypes.func.isRequired,
  viewHandler: PropTypes.func.isRequired,
  editHandler: PropTypes.func.isRequired,
  deleteHandler: PropTypes.func.isRequired,
};

export default CharacterProfileCard;
