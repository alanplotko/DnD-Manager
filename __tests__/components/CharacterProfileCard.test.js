import React from 'react';
import PropTypes from 'prop-types';
import CharacterProfileCard from 'FifthEditionManager/components/CharacterProfileCard';
import { View } from 'react-native';
import Modal from 'react-native-modal';
import { EXPERIENCE } from 'FifthEditionManager/config/Info';
import { cloneDeep } from 'lodash';

// Define wrapper component for testing interaction with child CharacterProfileCard component
class CharacterProfileCardWrapper extends React.Component {
  static propTypes = {
    character: PropTypes.object.isRequired,
    viewHandler: PropTypes.func.isRequired,
    editHandler: PropTypes.func.isRequired,
    deleteHandler: PropTypes.func.isRequired,
  }

  constructor(props) {
    super(props);
    this.state = {
      character: props.character,
      modalContent: null,
      isModalVisible: false,
      modalHandler: modalContent => this.setState({ modalContent, isModalVisible: true }),
      viewHandler: props.viewHandler,
      editHandler: props.editHandler,
      deleteHandler: props.deleteHandler,
    };
  }

  render() {
    return (
      <View>
        <CharacterProfileCard {...this.state} />
        <Modal
          isVisible={this.state.isModalVisible}
          onBackButtonPress={() => this.setState({ isModalVisible: false })}
        >
          <View>{this.state.modalContent}</View>
        </Modal>
      </View>
    );
  }
}

describe('Character Profile Card', () => {
  const character = {
    key: 'key',
    meta: {
      lastUpdated: 0,
    },
    race: { lookupKey: 'Race', name: 'Race' },
    baseClass: { lookupKey: 'Class', name: 'Class' },
    background: { lookupKey: 'Background', name: 'Background' },
    profile: {
      firstName: 'First',
      lastName: 'Last',
      level: 1,
      experience: 0,
    },
  };
  let modalStub;
  let viewStub;
  let editStub;
  let deleteStub;
  let timeStub;

  beforeEach(() => {
    // Use consistent date for tests
    timeStub = sinon.stub(Date, 'now').returns(0);

    modalStub = sinon.stub();
    viewStub = sinon.stub();
    editStub = sinon.stub();
    deleteStub = sinon.stub();
  });

  afterEach(() => {
    timeStub.restore();
    modalStub.reset();
    viewStub.reset();
    editStub.reset();
    deleteStub.reset();
  });

  test('renders properly for characters below max level', () => {
    const wrapper = shallow(<CharacterProfileCard
      character={character}
      modalHandler={modalStub}
      viewHandler={viewStub}
      editHandler={editStub}
      deleteHandler={deleteStub}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  test('renders properly for characters at max level', () => {
    // Override default character to have max level
    const characterMaxLevel = cloneDeep(character);
    characterMaxLevel.profile.level = EXPERIENCE.length;
    characterMaxLevel.profile.experience = EXPERIENCE[EXPERIENCE.length - 1];

    const wrapper = shallow(<CharacterProfileCard
      character={characterMaxLevel}
      modalHandler={modalStub}
      viewHandler={viewStub}
      editHandler={editStub}
      deleteHandler={deleteStub}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  test('can toggle the modal', () => {
    const wrapper = shallow(<CharacterProfileCard
      character={character}
      modalHandler={modalStub}
      viewHandler={viewStub}
      editHandler={editStub}
      deleteHandler={deleteStub}
    />);

    // 'More options' icon not pressed
    expect(modalStub.notCalled).toBe(true);

    // User must press the 'more options' icon to open the modal
    wrapper.find('IconToggle').props().onPress();

    // Modal should be open
    expect(modalStub.calledOnce).toBe(true);
  });

  test('can select all options in the modal', () => {
    const wrapper = shallow(<CharacterProfileCardWrapper
      character={character}
      viewHandler={viewStub}
      editHandler={editStub}
      deleteHandler={deleteStub}
    />);

    // No options selected
    expect(viewStub.notCalled).toBe(true);
    expect(editStub.notCalled).toBe(true);
    expect(deleteStub.notCalled).toBe(true);

    // Modal should be closed
    expect(wrapper).toMatchSnapshot();

    // User must press the 'more options' icon to open the modal
    wrapper.find('CharacterProfileCard').dive().find('IconToggle').props().onPress();
    wrapper.update();

    // Modal should be open
    expect(wrapper).toMatchSnapshot();

    // Modal should have 4 rows (character name and the three options: view, edit, delete)
    expect(wrapper.find('ReactNativeModal').children().find('ListItem')).toHaveLength(4);

    // Option handlers should execute on press
    wrapper.find('ReactNativeModal').children().find('ListItem').at(1).props().onPress();
    expect(viewStub.calledOnce).toBe(true);
    wrapper.find('ReactNativeModal').children().find('ListItem').at(2).props().onPress();
    expect(editStub.calledOnce).toBe(true);
    wrapper.find('ReactNativeModal').children().find('ListItem').at(3).props().onPress();
    expect(deleteStub.calledOnce).toBe(true);
  });
});
