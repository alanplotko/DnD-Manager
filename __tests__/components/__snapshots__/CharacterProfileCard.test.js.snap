// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character Profile Card can select all options in the modal 1`] = `
<View>
  <CharacterProfileCard
    character={
      Object {
        "background": Object {
          "lookupKey": "Background",
          "name": "Background",
        },
        "baseClass": Object {
          "lookupKey": "Class",
          "name": "Class",
        },
        "key": "key",
        "meta": Object {
          "lastUpdated": 0,
        },
        "profile": Object {
          "experience": 0,
          "firstName": "First",
          "lastName": "Last",
          "level": 1,
        },
        "race": Object {
          "lookupKey": "Race",
          "name": "Race",
        },
      }
    }
    deleteHandler={[Function]}
    editHandler={[Function]}
    isModalVisible={false}
    modalContent={null}
    modalHandler={[Function]}
    uiTheme={
      Object {
        "palette": Object {
          "accentColor": "#f44336",
          "backdropIconColor": "#bdbdbd",
          "backgroundColor": "#eeeeee",
          "disabledColor": "#e0e0e0",
          "fadedBackgroundColor": "rgba(0, 0, 0, 0.7)",
          "fadedTextColor": "#616161",
          "highlightedDiceColor": "#f44336",
          "iconColor": "#757575",
          "modalBackgroundColor": "#ffffff",
          "noteColor": "#9e9e9e",
          "primaryColor": "#3f51b5",
          "standardDiceColor": "#424242",
          "textColor": "#000000",
        },
        "toolbar": Object {
          "container": Object {
            "elevation": 0,
            "shadowOffset": Object {
              "height": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          },
        },
      }
    }
    viewHandler={[Function]}
  />
  <ReactNativeModal
    animationIn="slideInUp"
    animationInTiming={300}
    animationOut="slideOutDown"
    animationOutTiming={300}
    avoidKeyboard={false}
    backdropColor="black"
    backdropOpacity={0.7}
    backdropTransitionInTiming={300}
    backdropTransitionOutTiming={300}
    hideModalContentWhileAnimating={false}
    isVisible={false}
    onBackButtonPress={[Function]}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalShow={[Function]}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    swipeThreshold={100}
    useNativeDriver={false}
  >
    <View />
  </ReactNativeModal>
</View>
`;

exports[`Character Profile Card can select all options in the modal 2`] = `
<View>
  <CharacterProfileCard
    character={
      Object {
        "background": Object {
          "lookupKey": "Background",
          "name": "Background",
        },
        "baseClass": Object {
          "lookupKey": "Class",
          "name": "Class",
        },
        "key": "key",
        "meta": Object {
          "lastUpdated": 0,
        },
        "profile": Object {
          "experience": 0,
          "firstName": "First",
          "lastName": "Last",
          "level": 1,
        },
        "race": Object {
          "lookupKey": "Race",
          "name": "Race",
        },
      }
    }
    deleteHandler={[Function]}
    editHandler={[Function]}
    isModalVisible={true}
    modalContent={
      <View
        style={
          Object {
            "margin": 0,
          }
        }
      >
        <ListItem
          centerElement={
            <Styled(Text)
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "Roboto",
                    "fontSize": 14,
                  },
                  Object {
                    "color": "#000000",
                  },
                ]
              }
            >
              First Last
            </Styled(Text)>
          }
          dense={false}
          divider={false}
          leftElement={
            <Image
              resizeMode="contain"
              style={
                Object {
                  "height": 36,
                  "width": 36,
                }
              }
            />
          }
          numberOfLines={1}
          onLeftElementPress={null}
          onLongPress={null}
          onPress={null}
          onPressValue={null}
          onRightElementPress={null}
          rightElement={null}
          style={
            Object {
              "container": Object {
                "borderBottomColor": "#e0e0e0",
                "borderBottomWidth": 1,
                "height": 55,
                "margin": 0,
              },
              "contentViewContainer": Object {
                "backgroundColor": "#fafafa",
              },
              "leftElementContainer": Object {
                "marginRight": -10,
              },
            }
          }
          testID={null}
        />
        <ListItem
          centerElement={
            <Styled(Text)
              style={
                Object {
                  "color": "#000000",
                  "fontFamily": "Roboto",
                  "fontSize": 14,
                }
              }
            >
              View Character
            </Styled(Text)>
          }
          dense={false}
          divider={false}
          leftElement={
            <Icon
              color="#757575"
              name="open-in-new"
            />
          }
          numberOfLines={1}
          onLeftElementPress={null}
          onLongPress={null}
          onPress={[Function]}
          onPressValue={null}
          onRightElementPress={null}
          rightElement={null}
          style={
            Object {
              "container": Object {
                "height": 50,
                "margin": 0,
              },
              "leftElementContainer": Object {
                "marginRight": -10,
              },
            }
          }
          testID={null}
        />
        <ListItem
          centerElement={
            <Styled(Text)
              style={
                Object {
                  "color": "#000000",
                  "fontFamily": "Roboto",
                  "fontSize": 14,
                }
              }
            >
              Edit Character
            </Styled(Text)>
          }
          dense={false}
          divider={false}
          leftElement={
            <Icon
              color="#757575"
              name="mode-edit"
            />
          }
          numberOfLines={1}
          onLeftElementPress={null}
          onLongPress={null}
          onPress={[Function]}
          onPressValue={null}
          onRightElementPress={null}
          rightElement={null}
          style={
            Object {
              "container": Object {
                "height": 50,
                "margin": 0,
              },
              "leftElementContainer": Object {
                "marginRight": -10,
              },
            }
          }
          testID={null}
        />
        <ListItem
          centerElement={
            <Styled(Text)
              style={
                Object {
                  "color": "#000000",
                  "fontFamily": "Roboto",
                  "fontSize": 14,
                }
              }
            >
              Delete Character
            </Styled(Text)>
          }
          dense={false}
          divider={false}
          leftElement={
            <Icon
              color="#757575"
              name="delete"
            />
          }
          numberOfLines={1}
          onLeftElementPress={null}
          onLongPress={null}
          onPress={[Function]}
          onPressValue={null}
          onRightElementPress={null}
          rightElement={null}
          style={
            Object {
              "container": Object {
                "height": 50,
                "margin": 0,
              },
              "leftElementContainer": Object {
                "marginRight": -10,
              },
            }
          }
          testID={null}
        />
      </View>
    }
    modalHandler={[Function]}
    uiTheme={
      Object {
        "palette": Object {
          "accentColor": "#f44336",
          "backdropIconColor": "#bdbdbd",
          "backgroundColor": "#eeeeee",
          "disabledColor": "#e0e0e0",
          "fadedBackgroundColor": "rgba(0, 0, 0, 0.7)",
          "fadedTextColor": "#616161",
          "highlightedDiceColor": "#f44336",
          "iconColor": "#757575",
          "modalBackgroundColor": "#ffffff",
          "noteColor": "#9e9e9e",
          "primaryColor": "#3f51b5",
          "standardDiceColor": "#424242",
          "textColor": "#000000",
        },
        "toolbar": Object {
          "container": Object {
            "elevation": 0,
            "shadowOffset": Object {
              "height": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          },
        },
      }
    }
    viewHandler={[Function]}
  />
  <ReactNativeModal
    animationIn="slideInUp"
    animationInTiming={300}
    animationOut="slideOutDown"
    animationOutTiming={300}
    avoidKeyboard={false}
    backdropColor="black"
    backdropOpacity={0.7}
    backdropTransitionInTiming={300}
    backdropTransitionOutTiming={300}
    hideModalContentWhileAnimating={false}
    isVisible={true}
    onBackButtonPress={[Function]}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalShow={[Function]}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    swipeThreshold={100}
    useNativeDriver={false}
  >
    <View>
      <View
        style={
          Object {
            "margin": 0,
          }
        }
      >
        <ListItem
          centerElement={
            <Styled(Text)
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "Roboto",
                    "fontSize": 14,
                  },
                  Object {
                    "color": "#000000",
                  },
                ]
              }
            >
              First Last
            </Styled(Text)>
          }
          dense={false}
          divider={false}
          leftElement={
            <Image
              resizeMode="contain"
              style={
                Object {
                  "height": 36,
                  "width": 36,
                }
              }
            />
          }
          numberOfLines={1}
          onLeftElementPress={null}
          onLongPress={null}
          onPress={null}
          onPressValue={null}
          onRightElementPress={null}
          rightElement={null}
          style={
            Object {
              "container": Object {
                "borderBottomColor": "#e0e0e0",
                "borderBottomWidth": 1,
                "height": 55,
                "margin": 0,
              },
              "contentViewContainer": Object {
                "backgroundColor": "#fafafa",
              },
              "leftElementContainer": Object {
                "marginRight": -10,
              },
            }
          }
          testID={null}
        />
        <ListItem
          centerElement={
            <Styled(Text)
              style={
                Object {
                  "color": "#000000",
                  "fontFamily": "Roboto",
                  "fontSize": 14,
                }
              }
            >
              View Character
            </Styled(Text)>
          }
          dense={false}
          divider={false}
          leftElement={
            <Icon
              color="#757575"
              name="open-in-new"
            />
          }
          numberOfLines={1}
          onLeftElementPress={null}
          onLongPress={null}
          onPress={[Function]}
          onPressValue={null}
          onRightElementPress={null}
          rightElement={null}
          style={
            Object {
              "container": Object {
                "height": 50,
                "margin": 0,
              },
              "leftElementContainer": Object {
                "marginRight": -10,
              },
            }
          }
          testID={null}
        />
        <ListItem
          centerElement={
            <Styled(Text)
              style={
                Object {
                  "color": "#000000",
                  "fontFamily": "Roboto",
                  "fontSize": 14,
                }
              }
            >
              Edit Character
            </Styled(Text)>
          }
          dense={false}
          divider={false}
          leftElement={
            <Icon
              color="#757575"
              name="mode-edit"
            />
          }
          numberOfLines={1}
          onLeftElementPress={null}
          onLongPress={null}
          onPress={[Function]}
          onPressValue={null}
          onRightElementPress={null}
          rightElement={null}
          style={
            Object {
              "container": Object {
                "height": 50,
                "margin": 0,
              },
              "leftElementContainer": Object {
                "marginRight": -10,
              },
            }
          }
          testID={null}
        />
        <ListItem
          centerElement={
            <Styled(Text)
              style={
                Object {
                  "color": "#000000",
                  "fontFamily": "Roboto",
                  "fontSize": 14,
                }
              }
            >
              Delete Character
            </Styled(Text)>
          }
          dense={false}
          divider={false}
          leftElement={
            <Icon
              color="#757575"
              name="delete"
            />
          }
          numberOfLines={1}
          onLeftElementPress={null}
          onLongPress={null}
          onPress={[Function]}
          onPressValue={null}
          onRightElementPress={null}
          rightElement={null}
          style={
            Object {
              "container": Object {
                "height": 50,
                "margin": 0,
              },
              "leftElementContainer": Object {
                "marginRight": -10,
              },
            }
          }
          testID={null}
        />
      </View>
    </View>
  </ReactNativeModal>
</View>
`;

exports[`Character Profile Card renders properly for characters at max level 1`] = `
<Styled(Card)
  style={
    Object {
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 10,
    }
  }
>
  <Styled(CardItem)
    cardBody={true}
  >
    <Image
      style={
        Object {
          "flex": 1,
          "height": 100,
          "width": null,
        }
      }
    />
    <View
      style={
        Object {
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <IconToggle
        color="#ffffff"
        name="more-vert"
        onPress={[Function]}
      />
    </View>
    <View
      style={
        Object {
          "bottom": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <Styled(Text)
        style={
          Array [
            Object {
              "borderTopLeftRadius": 5,
              "color": "#ffffff",
              "fontFamily": "Roboto",
              "fontSize": 12,
              "fontStyle": "italic",
              "paddingBottom": 3,
              "paddingLeft": 5,
              "paddingRight": 5,
              "paddingTop": 3,
            },
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.7)",
            },
          ]
        }
      >
        Last updated 
        a few seconds ago
      </Styled(Text)>
    </View>
  </Styled(CardItem)>
  <Styled(CardItem)>
    <Styled(Left)
      style={
        Object {
          "flex": 2,
        }
      }
    >
      <Image
        resizeMode="contain"
        style={
          Object {
            "height": 64,
            "width": 48,
          }
        }
      />
      <Styled(Body)>
        <Styled(Text)
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "RobotoLight",
                "fontSize": 18,
                "width": 175,
              },
              Object {
                "color": "#000000",
              },
            ]
          }
        >
          First Last
           
        </Styled(Text)>
        <Styled(Text)
          style={
            Array [
              Object {
                "color": "#9e9e9e",
                "fontFamily": "Roboto",
                "fontSize": 14,
              },
              Object {
                "color": "#9e9e9e",
              },
            ]
          }
        >
          Race
           
          Class
        </Styled(Text)>
        <Styled(Text)
          style={
            Array [
              Object {
                "color": "#9e9e9e",
                "fontFamily": "Roboto",
                "fontSize": 14,
              },
              Object {
                "color": "#9e9e9e",
              },
            ]
          }
        >
          Background
        </Styled(Text)>
      </Styled(Body)>
    </Styled(Left)>
    <Styled(Right)
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Styled(Text)
          style={
            Array [
              Object {
                "color": "#9e9e9e",
                "fontFamily": "RobotoLight",
                "fontSize": 16,
              },
              Object {
                "color": "#9e9e9e",
              },
            ]
          }
        >
          Level 
          20
        </Styled(Text)>
        <ProgressBar
          animated={false}
          animationConfig={
            Object {
              "bounciness": 0,
            }
          }
          animationType="spring"
          borderColor="#3f51b5"
          borderRadius={4}
          borderWidth={1}
          color="#3f51b5"
          height={6}
          indeterminate={false}
          progress={1}
          style={
            Object {
              "marginBottom": 3,
              "marginTop": 3,
            }
          }
          useNativeDriver={false}
          width={100}
        />
        <Styled(Text)
          style={
            Object {
              "color": "#9e9e9e",
              "fontFamily": "RobotoLight",
              "fontSize": 14,
            }
          }
        >
          355.0k
        </Styled(Text)>
      </View>
    </Styled(Right)>
  </Styled(CardItem)>
</Styled(Card)>
`;

exports[`Character Profile Card renders properly for characters below max level 1`] = `
<Styled(Card)
  style={
    Object {
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 10,
    }
  }
>
  <Styled(CardItem)
    cardBody={true}
  >
    <Image
      style={
        Object {
          "flex": 1,
          "height": 100,
          "width": null,
        }
      }
    />
    <View
      style={
        Object {
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <IconToggle
        color="#ffffff"
        name="more-vert"
        onPress={[Function]}
      />
    </View>
    <View
      style={
        Object {
          "bottom": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <Styled(Text)
        style={
          Array [
            Object {
              "borderTopLeftRadius": 5,
              "color": "#ffffff",
              "fontFamily": "Roboto",
              "fontSize": 12,
              "fontStyle": "italic",
              "paddingBottom": 3,
              "paddingLeft": 5,
              "paddingRight": 5,
              "paddingTop": 3,
            },
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.7)",
            },
          ]
        }
      >
        Last updated 
        a few seconds ago
      </Styled(Text)>
    </View>
  </Styled(CardItem)>
  <Styled(CardItem)>
    <Styled(Left)
      style={
        Object {
          "flex": 2,
        }
      }
    >
      <Image
        resizeMode="contain"
        style={
          Object {
            "height": 64,
            "width": 48,
          }
        }
      />
      <Styled(Body)>
        <Styled(Text)
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "RobotoLight",
                "fontSize": 18,
                "width": 175,
              },
              Object {
                "color": "#000000",
              },
            ]
          }
        >
          First Last
           
        </Styled(Text)>
        <Styled(Text)
          style={
            Array [
              Object {
                "color": "#9e9e9e",
                "fontFamily": "Roboto",
                "fontSize": 14,
              },
              Object {
                "color": "#9e9e9e",
              },
            ]
          }
        >
          Race
           
          Class
        </Styled(Text)>
        <Styled(Text)
          style={
            Array [
              Object {
                "color": "#9e9e9e",
                "fontFamily": "Roboto",
                "fontSize": 14,
              },
              Object {
                "color": "#9e9e9e",
              },
            ]
          }
        >
          Background
        </Styled(Text)>
      </Styled(Body)>
    </Styled(Left)>
    <Styled(Right)
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Styled(Text)
          style={
            Array [
              Object {
                "color": "#9e9e9e",
                "fontFamily": "RobotoLight",
                "fontSize": 16,
              },
              Object {
                "color": "#9e9e9e",
              },
            ]
          }
        >
          Level 
          1
        </Styled(Text)>
        <ProgressBar
          animated={false}
          animationConfig={
            Object {
              "bounciness": 0,
            }
          }
          animationType="spring"
          borderColor="#3f51b5"
          borderRadius={4}
          borderWidth={1}
          color="#3f51b5"
          height={6}
          indeterminate={false}
          progress={0}
          style={
            Object {
              "marginBottom": 3,
              "marginTop": 3,
            }
          }
          useNativeDriver={false}
          width={100}
        />
        <Styled(Text)
          style={
            Object {
              "color": "#9e9e9e",
              "fontFamily": "RobotoLight",
              "fontSize": 14,
            }
          }
        >
          0
           / 
          300
        </Styled(Text)>
      </View>
    </Styled(Right)>
  </Styled(CardItem)>
</Styled(Card)>
`;
